addons/fleet/models/fleet_vehicle.py

```diff
@@ -202,10 +202,16 @@ def _compute_contract_reminder(self):
             total = 0
             name = ''
             state = ''
-            for element in record.log_contracts:
-                if element.state in ('open', 'expired') and element.expiration_date:
+
+             # Get the last contract by ordering by creation date
+            log_contract = self.env['fleet.vehicle.log.contract'].search([
+                ('vehicle_id', '=', record.id)
+                ], limit=1, order='create_date desc')
+
+            if log_contract and log_contract.expiration_date:
+                if log_contract.state in ('open', 'expired'):
```

Comment:
Is there a reason you do this if-check after determining log_contract and not when searching for the log_contract? What happens when the last created contract is a cancelled contract. It seems to me that in that case there will never be an overdue contract because this if-branch is never entered, even though there still might be.

Reply:
No, not really, I just coded in a very iterative way and not refactored my code enough I will correct that!